index.js

```diff
@@ -139,21 +139,24 @@ var Field = redefine.Class({
 });
 
 var SearchResult = redefine.Class({
-  constructor: function SearchResult() {},
+  constructor: function SearchResult(ItemType, object) {
+    this.responseData = object;
+    this.total = object.total;
+    this.limit = object.limit;
+    this.skip = object.skip;
+    walkMutate(_.cloneDeep(object), isParseableResource, parseResource);
+    resolveResponse(object).forEach(function (item, i) {
+      this[i] = item;
```

Comment:
why are you assigning in the SearchResult instance every item to a property named after its index?